RecoTracker/TkSeedGenerator/plugins/ClusterCheckerEDProducer.cc

```diff
@@ -0,0 +1,55 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "RecoTracker/TkSeedGenerator/interface/ClusterChecker.h"
+
+class ClusterCheckerEDProducer: public edm::stream::EDProducer<> {
+public:
+
+  ClusterCheckerEDProducer(const edm::ParameterSet& iConfig);
+  ~ClusterCheckerEDProducer() = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  virtual void produce(edm::Event& iEvent, const edm::EventSetup& iSetup) override;
+
+private:
+  ClusterChecker theClusterCheck;
+  bool theSilentOnClusterCheck;
+};
+
+ClusterCheckerEDProducer::ClusterCheckerEDProducer(const edm::ParameterSet& iConfig):
+  theClusterCheck(iConfig, consumesCollector()),
+  theSilentOnClusterCheck(iConfig.getUntrackedParameter<bool>("silentClusterCheck"))
+{
+  produces<bool>();
+}
+
+void ClusterCheckerEDProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  ClusterChecker::fillDescriptions(desc);
+  desc.addUntracked<bool>("silentClusterCheck", false);
+
+  descriptions.add("trackerClusterCheck", desc);
+}
+
+void ClusterCheckerEDProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  auto ret = std::make_unique<bool>(true);
+
+  //protection for big ass events...
+  size_t clustsOrZero = theClusterCheck.tooManyClusters(iEvent);
+  if (clustsOrZero){
+    if (!theSilentOnClusterCheck)
+	edm::LogError("TooManyClusters") << "Found too many clusters (" << clustsOrZero << "), bailing out.\n";
```

Comment:
The current message has the newline https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromRegionHitsEDProducer.cc#L87.

@VinInn, @rovere shall I remove the trailing newline?